src/json.rs

```diff
@@ -201,13 +207,44 @@ impl<S> JsonConfig<S> {
         self.ehandler = Rc::new(f);
         self
     }
+
+    /// Add allowed content type
+    pub fn content_type(&mut self, content_type: ContentTypeFilter) -> &mut Self {
+        self.content_types.push(content_type);
+        self
+    }
 }
 
 impl<S> Default for JsonConfig<S> {
     fn default() -> Self {
+        let mut content_types = Vec::new();
+        content_types.push(ContentTypeFilter::new(None, Some(mime::JSON), None));
+        content_types.push(ContentTypeFilter::new(None, None, Some(mime::JSON)));
         JsonConfig {
             limit: 262_144,
             ehandler: Rc::new(|e, _| e.into()),
+            content_types,
+        }
+    }
+}
+
+/// Filter type for allowed `Content-Type` header values.
+pub struct ContentTypeFilter {
+    type_: Option<mime::Name<'static>>,
```

Comment:
I don't think it is convenient to use struct with lots of optional fields like that.

I think it might be better if user would be able to supply `mime::Mime` rather than allowing that accurate checks.
On side note it might be better if user could provide function that takes `mime::Mime` and returns whether it should be treated as json or not.